### PR TITLE
feat: add generated .NET 6 heads from UnoPlatformApp

### PR DESCRIPTION
**Pull request by GabrielChouinardLetourneau**
GitHub Issue: #

## PR Type

What kind of change does this PR introduce?

- Feature
- Build or CI related changes

## Description

Theses new project have been added **without any changes made to them**, a shared generated by Uno. More pipeline changes will be committed to publish the new blank app.

**Update:** The pipeline builds the platform with the exception of iOS when built on a Mac agent. I had to use a hosted agent instead because I had a permission issue when running the installation of dotnet 6 workloads.

## PR Checklist 
Please check if your PR fulfills the following requirements:

- [ ] Interface members are XML documented
- [ ] Documentation (XML or comments) has been added and/or existing documentation has been updated
- [ ] [Architecture documents](./doc/Architecture.md) have been updated
- [ ] Tested on all relevant platforms

### CODE REVIEW

File: src/app/ApplicationTemplate.Mobile/Android/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.companyname.ApplicationTemplate" android:versionCode="1" android:versionName="1.0">
+  <uses-sdk android:minSdkVersion="21" android:targetSdkVersion="31" />
```
Comment (carlh98) [inline src/app/ApplicationTemplate.Mobile/Android/AndroidManifest.xml:3]: You can increase the min sdk to 26

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 2]
```diff
@@ -0,0 +1,46 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<PropertyGroup>
+		<TargetFrameworks>net6.0-android</TargetFrameworks>
+		<TargetFrameworks>$(TargetFrameworks);net6.0-ios</TargetFrameworks>
+		<SingleProject>true</SingleProject>
+		<OutputType>Exe</OutputType>
+		<IsUnoHead>true</IsUnoHead>
+		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">14.2</SupportedOSPlatformVersion>
+		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-android'">21.0</SupportedOSPlatformVersion>
```
Comment (carlh98) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:9]: Same here increase to 26

**Comment by carlh98** [comment]
Please mac and mac catalyst support

File: .azure-pipelines.yml [hunk 3]
```diff
@@ -30,32 +30,8 @@ variables:
 - template: build/variables.yml
 
 stages:
-#-if false
```
Comment (jeanplevesque) [inline .azure-pipelines.yml:33]: Why would you remove this?

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

File: .azure-pipelines.yml (hunk 3 above)
Comment (jeanplevesque) [reply .azure-pipelines.yml:33]: This section tests that the dotnet new command works and generates an app that actually builds.

File: build/stage-build.yml [hunk 4]
```diff
@@ -75,44 +70,103 @@ jobs:
...
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: ${{ parameters.solutionFileName }}
+
+
+  - job: OnWindows_Android
+    strategy:
+      maxParallel: 3
+      matrix:
+        Android_AppCenter:
```
Comment (jeanplevesque) [inline build/stage-build.yml:84]: Why is this called Android**_AppCenter**?

File: build/templates/dotnet-install-mac.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      installationPath: ${{ parameters.Dotnet_Root }}
+
+  - task: UseDotNet@2
+    displayName: install .NET 3.1
+    retryCountOnTaskFailure: 3
+    inputs:
+      version: 3.1.x
+      installationPath: ${{ parameters.Dotnet_Root }}
+
+  # Required until .NET 6 installs properly using UseDotnet
```
Comment (jeanplevesque) [inline build/templates/dotnet-install-mac.yml:23]: Can this be removed?

File: src/app/ApplicationTemplate.Droid/ApplicationTemplate.Droid.csproj [hunk 6]
```diff
@@ -73,6 +73,20 @@
...
+    <DefineConstants>TRACE</DefineConstants>
+    <Optimize>true</Optimize>
+    <DebugType>portable</DebugType>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <GenerateSerializationAssemblies>Off</GenerateSerializationAssemblies>
+    <LangVersion>10.0</LangVersion>
+    <ErrorReport>prompt</ErrorReport>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release_net6-ios|AnyCPU'">
+    <OutputPath>bin\Release_net6-ios\</OutputPath>
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Droid/ApplicationTemplate.Droid.csproj:88]: I don't understand why this Xamarin.Androind csproj is modified with Windows and ios specific instructions?

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 7]
```diff
@@ -0,0 +1,46 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<PropertyGroup>
+		<TargetFrameworks>net6.0-ios</TargetFrameworks>
+		<TargetFrameworks>$(TargetFrameworks);net6.0-android</TargetFrameworks>
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:4]: Couldn't we just put both on the same line?

File: src/app/ApplicationTemplate.UWP/ApplicationTemplate.UWP.csproj [hunk 8]
```diff
@@ -98,6 +98,57 @@
     <UseDotNetNativeToolchain>false</UseDotNetNativeToolchain>
     <DocumentationFile>bin\x64\Release\Documentation.xml</DocumentationFile>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release_Windows|x86'">
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.UWP/ApplicationTemplate.UWP.csproj:101]: I don't understand these changes.

File: src/app/ApplicationTemplate.Tests/ApplicationTemplate.Tests.csproj [hunk 9]
```diff
@@ -7,6 +7,7 @@
 		<LangVersion>10.0</LangVersion>
 		<IsTestProject>True</IsTestProject>
 		<DefineConstants>NETFRAMEWORK</DefineConstants>
+		<Configurations>Debug;Release;Release_Windows;Release_net6-ios</Configurations>
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Tests/ApplicationTemplate.Tests.csproj:10]: I don't understand these changes.

File: src/app/ApplicationTemplate.UITests/ApplicationTemplate.UITests.csproj [hunk 10]
```diff
@@ -4,6 +4,7 @@
 		<DebugType>full</DebugType>
 		<DebugSymbols>true</DebugSymbols>
 		<LangVersion>10.0</LangVersion>
+		<Configurations>Debug;Release;Release_Windows;Release_net6-ios</Configurations>
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.UITests/ApplicationTemplate.UITests.csproj:7]: I don't understand these changes.

File: src/app/ApplicationTemplate.iOS/ApplicationTemplate.iOS.csproj [hunk 11]
```diff
@@ -66,6 +66,35 @@
     <BuildIpa>True</BuildIpa>
     <CodesignKey>iPhone Distribution</CodesignKey>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release_Windows|iPhoneSimulator'">
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.iOS/ApplicationTemplate.iOS.csproj:69]: I don't understand these changes.

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

**Review by jeanplevesque — DISMISSED** [review]

File: src/app/ApplicationTemplate.Business/ApplicationTemplate.Business.csproj [hunk 12]
```diff
@@ -3,6 +3,7 @@
 	<PropertyGroup>
 		<TargetFramework>netstandard2.0</TargetFramework>
 		<LangVersion>10.0</LangVersion>
+		<Configurations>Debug;Release;Release_Windows;Release_net6-ios</Configurations>
```
[the remark this answers is not in the document]
Comment (Soap-141) [reply src/app/ApplicationTemplate.Business/ApplicationTemplate.Business.csproj]: I asked Gabriel and it seems that they were automatically added when he made change in the configuration manager. But to be honest I don't even know why we are messing with the configuration manager.

File: build/stage-build.yml [hunk 13]
```diff
@@ -75,44 +70,102 @@ jobs:
     pool:
       name: $(windowsPoolName)
 
+    steps:
+      - template: steps-build-windows.yml
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: ${{ parameters.solutionFileName }}
+
```
Comment (carlh98) [inline build/stage-build.yml:78]: You can remove the added line

File: build/stage-build.yml [hunk 14]
```diff
@@ -75,44 +70,102 @@ jobs:
...
+        versionCode: $(PreReleaseNumber)
+        versionName: $(MajorMinorPatch)
+        printFile: true
+
+    - template: steps-build-android.yml
+
+  - job: OnMac_iOS
     strategy:
-      maxParallel: 3
+      maxParallel: 2
```
Comment (carlh98) [inline build/stage-build.yml:124]: @Soap-141 Since there's only one, we can remove the maxParallel right?

File: build/steps-build-ios.yml [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  retryCountOnTaskFailure: 3
+  inputs:
+    versionSpec: '5.10.1'
+  displayName: 'Install GitVersion'
+
+- task: gitversion/execute@0
+  inputs:
+    useConfigFile: true
+    configFilePath: $(Build.SourcesDirectory)/gitversion.yml
+  displayName: 'Calculate version'
```
Comment (carlh98) [inline build/steps-build-ios.yml:20]: We don't need to install the dotnet 6 workloads?

File: build/stage-build.yml (hunk 14 above)
Comment (Soap-141) [reply build/stage-build.yml:124]: Didn't event know what it was. I can try.

Comment (carlh98) [reply build/stage-build.yml:124]: It's great for the win machine when we build android, uwp/windows and test at the same time :)

File: build/steps-build-android.yml [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+  inputs:
+    SourceFolder: $(build.sourcesdirectory)/src/app/ApplicationTemplate.Mobile/bin/Release/net6.0-android
+    Contents: |
+      **/*.aab
+      **/*.apk
+    TargetFolder: $(build.artifactstagingdirectory)
+    CleanTargetFolder: false
+    OverWrite: false
+    flattenFolders: false
+
```
Comment (carlh98) [inline build/steps-build-android.yml:46]: remove added line

File: build/stage-donetnew.yaml [hunk 17]
```diff
@@ -9,8 +9,8 @@ jobs:
       displayName: 'Install UnoApplicationTemplate'
 
     - script: dotnet new nv-mobile -n GeneratedApp
-      displayName: 'Generate project using dotnet new'
+      displayName: 'Generate Project'
 
     - publish: GeneratedApp
-      displayName: 'Publish GeneratedApp folder'
+      displayName: 'Publish GeneratedApp fFlder'
```
Comment (carlh98) [inline build/stage-donetnew.yaml:15]: typo

File: build/stage-build.yml (hunk 14 above)
Comment (carlh98) [reply build/stage-build.yml:108]: @Soap-141  You can remove maxParallel for mac since it's not needed :)

File: build/stage-build.yml [hunk 18]
```diff
@@ -39,80 +42,95 @@
...
+    pool:
+      vmImage : windows-2022
+    steps:
+      - template: steps-build-tests.yml
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: '${{ parameters.solutionName }}.sln'
+          solutionName: ${{ parameters.solutionName }}
+
+  - job: OnWindows_Android
```
Comment (vincentcastagna) [inline build/stage-build.yml:87]: As this job can take more than an hour to complete (due to AOT, and probably we need to improve that), you can set a higher limit than the default 60 minutes through `timeoutInMinutes`.

```yml
- job: OnWindows_Android
  timeoutInMinutes: 90             
  strategy:
    matrix:
      Android:
        ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
        ArtifactName: $(AndroidArtifactName)_${{ parameters.applicationEnvironment }}
        ApplicationPlatform: AnyCPU
        BuildForPlayStore: False
        BuildTargetFramework: net6.0-android
        VersionCodeOffset: 1
```

File: src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml [hunk 19]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.nventive.internal.applicationtemplate" android:versionCode="1" android:versionName="1.0" android:installLocation="auto">
-	<uses-sdk android:minSdkVersion="26" android:targetSdkVersion="31" />
+	<uses-sdk android:minSdkVersion="18" android:targetSdkVersion="31" />
```
Comment (carlh98) [inline src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml:3]: Curious about the downgrade to 18?

Comment (Soap-141) [reply src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml:3]: I don't remember doing that, good catch. But at the end of the day it doesn't matter this project will be removed.

File: build/steps-build-android.yml [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+- task: android-manifest-version@1
+  displayName: "Bump Android Manifest Version"
+  inputs:
+    sourcePath: ${{ parameters.pathToAndroidManifest }}
+    versionCode: $(PreReleaseNumber)
+    versionName: $(MajorMinorPatch)
+    printFile: true
+
+- script: >
```
Comment (vincentcastagna) [inline build/steps-build-android.yml:57]: Why do we have two scripts by the way ?

Comment (vincentcastagna) [reply build/steps-build-android.yml:57]: Nvm its for light build.

Comment (vincentcastagna) [reply build/steps-build-android.yml:57]: Still I don't get why you need to split ? Can't we make something to refacto this in a single script ?

Comment (vincentcastagna) [reply build/steps-build-android.yml:47]: Ok as discussed we can optimize that by dynamically set a variable for publish vs build.

Comment (Soap-141) [reply build/steps-build-android.yml:47]: Done! :)

File: build/stage-build.yml [hunk 21]
```diff
@@ -39,80 +42,100 @@
...
+    - template: steps-build-android.yml
+      parameters:
+        pathToSrc: ${{ parameters.pathToSrc }}
+        solutionName: ${{ parameters.solutionName }}
+        pathToAndroidManifest: ${{ parameters.pathToAndroidManifest }}
+        androidKeyStoreFile: ${{ parameters.androidKeyStoreFile }}
+
+  - job: OnMac_iOS
+    pool:
+      vmImage: macos-12
```
Comment (takla21) [inline build/stage-build.yml:115]: are we setting a specific agent here?

File: build/steps-build-android.yml [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+  default: ''
+- name: solutionName
+  type: string
+  default: ''
+- name: pathToAndroidManifest
+  type: string
+  default: ''
+
+steps:
+- template: templates/dotnet-install.yml
```
Comment (takla21) [inline build/steps-build-android.yml:16]: is it still necessary? I though we didn't need it

File: build/steps-build-windows.yml [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+#-if false
+- download: current
+  condition: eq('GeneratedApp.sln', '${{ parameters.solutionFileName }}')
+  artifact: GeneratedApp
+#-endif
+
+- script: >
+    cd ${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Windows
+
```
Comment (takla21) [inline build/steps-build-windows.yml:23]: why the empty line here?

File: src/app/ApplicationTemplate.Mobile/Android/Resources/values/Styles.xml [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+		<item name="windowActionBar">false</item>
+		<item name="android:windowActionBar">false</item>
+		<item name="windowNoTitle">true</item>
+		<item name="android:windowNoTitle">true</item>
+
+	</style>
+	<style name="Theme.AppCompat.Translucent">
+		<item name="android:windowIsTranslucent">true</item>
+		<item name="android:windowAnimationStyle">@android:style/Animation</item>
+	</style>
```
Comment (takla21) [inline src/app/ApplicationTemplate.Mobile/Android/Resources/values/Styles.xml:15]: I'm wondering if it should have the same content as here https://github.com/nventive/UnoApplicationTemplate/blob/main/src/app/ApplicationTemplate.Droid/Resources/values/Styles.xml

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 25]
```diff
@@ -0,0 +1,154 @@
...
+							<PropertyGroup>
+								<AndroidEnableProfiledAot>False</AndroidEnableProfiledAot>
+								<EnableLLVM>False</EnableLLVM>
+								<Optimize>False</Optimize>
+								<RunAOTCompilation>False</RunAOTCompilation>
+							</PropertyGroup>
+						</When>
+						<Otherwise>
+							<PropertyGroup>
+								<!-- Currently doesn't work with AOT enabled. -->
```
Comment (takla21) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:64]: Do we have a trace here for that? (should probably have a github issue in this repo as I don't think it's a good idea to reference anything private here)

File: build/steps-build-android.yml (hunk 22 above)
Comment (takla21) [reply build/steps-build-android.yml:16]: for android & winui

File: build/stage-build.yml (hunk 21 above)
Comment (Soap-141) [reply build/stage-build.yml:115]: No, we only need `vmImage` to use Microsoft hosted agent.

File: build/steps-build-android.yml (hunk 22 above)
Comment (Soap-141) [reply build/steps-build-android.yml:16]: Yes it is, it makes sure that we have the correct version of Dot Net installed and that we have the necessary mobile workloads.

File: build/steps-build-windows.yml (hunk 23 above)
Comment (Soap-141) [reply build/steps-build-windows.yml:23]: When we use `>` it converts line break to spaces, so to have an actual line break we need this.

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj (hunk 25 above)
Comment (Soap-141) [reply src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:64]: No nothing and I can't even have a log for this, using release mode and too much text in the logcat.

File: src/app/ApplicationTemplate.Mobile/Android/Resources/values/Styles.xml (hunk 24 above)
Comment (Soap-141) [reply src/app/ApplicationTemplate.Mobile/Android/Resources/values/Styles.xml:15]: May I could had [this](https://github.com/dotnet/runtime/issues/74021)?

File: build/steps-build-ios.yml [hunk 26]
```diff
@@ -0,0 +1,99 @@
...
+- task: CopyFiles@2
+  displayName: 'Copy DSYM'
+  inputs:
+    Contents: '**/*.dSYM/**'
+    flattenFolders : true
+    SourceFolder: '${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Mobile/bin/Release/net6.0-ios'
+    TargetFolder: '$(Build.ArtifactStagingDirectory)/${{ parameters.solutionName }}.Mobile.dSYM'
+
+- publish: $(Build.ArtifactStagingDirectory)
+  displayName: 'Publish Artifact $(ApplicationConfiguration) | $(ApplicationPlatform)'
```
Comment (vincentcastagna) [inline build/steps-build-ios.yml:83]: You don't pass in the `ApplicationConfiguration `leading in improper display in  the CI. 
You should pass it form the AzurePipelines.yml depending on the stage (Production etc ..) 

Same thing for `steps-build-android`

image

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj (hunk 25 above)
Comment (takla21) [reply src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:64]: yeah that's perfect

File: src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml (hunk 19 above)
Comment (Iguins) [inline src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml:3]: Minimum sdk supported should stay 26 and probably will be more higher next year.

File: src/app/ApplicationTemplate.Mobile/Android/AndroidManifest.xml [hunk 27]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Iguins) [inline src/app/ApplicationTemplate.Mobile/Android/AndroidManifest.xml:1]: If i'm not mistaken, we seems to have to androidManifest.xml. witch one is the correct one to be used?

Comment (Soap-141) [reply src/app/ApplicationTemplate.Mobile/Android/AndroidManifest.xml:1]: Mobile is the correct one, don't look at the Xamarin Android but just Mobile one.

File: src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml (hunk 19 above)
Comment (Soap-141) [reply src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml:3]: My bad, this project doesn't matter, it will be removed in the next step.

Comment (Soap-141) [reply src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml]: Fixed it to remove confusion.

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 28]
```diff
@@ -0,0 +1,154 @@
...
+	</ItemGroup>
+	<Choose>
+		<When Condition="'$(TargetFramework)'=='net6.0-android'">
+			<Choose>
+				<When Condition="'$(Configuration)'=='Debug'">
+					<PropertyGroup>
+						<AndroidUseSharedRuntime>True</AndroidUseSharedRuntime>
+						<UseHighDPIResources>False</UseHighDPIResources>
+						<EmbedAssembliesIntoApk>False</EmbedAssembliesIntoApk>
+						<AndroidCreatePackagePerAbi>False</AndroidCreatePackagePerAbi>
```
Comment (Iguins) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:33]: AndroidCreatePackagePerAbi seems to be repeated twice

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 29]
```diff
@@ -0,0 +1,154 @@
...
+							</PropertyGroup>
+						</Otherwise>
+					</Choose>
+				</When>
+			</Choose>
+			<ItemGroup>
+				<PackageReference Include="Uno.Extensions.Logging.OSLog" Version="1.4.0" />
+			</ItemGroup>
+			<ItemGroup>
+				<LinkDescription Include="iOS\LinkerExclusions.xml" />
```
Comment (Iguins) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:149]: The linker for iOS is present but seems to be missing for Android is the csproj

Comment (Soap-141) [reply src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:149]: That's right, I forgot. Will have to do some research because it don't know if it's useful.

Comment (Iguins) [reply src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:149]: It is useful for some packages we want the linker to ignore when there is not direct binding reference to it.

File: build/stage-build.yml [hunk 30]
```diff
@@ -39,80 +42,100 @@
...
-          ApplicationConfiguration: Release_iOS
-          ApplicationPlatform: Any CPU
-          ArtifactName: $(iOSArtifactName)_${{ parameters.applicationEnvironment }}
-          ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
+  - job: OnWindows_Android
     pool:
-      name: $(macOSPoolName)
-      demands:
-        - Xamarin.iOS_Version -equals $(XAMARIN_IOS_VERSION)
+      vmImage : windows-2022
```
Comment (Iguins) [inline build/stage-build.yml:85]: Same please use the same global variable

File: build/stage-build.yml [hunk 31]
```diff
@@ -39,80 +42,100 @@
...
+    steps:
+      - template: steps-build-windows.yml
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: '${{ parameters.solutionName }}.sln'
+          solutionName: ${{ parameters.solutionName }}
 
+  - job: OnWindows_Tests
+    pool:
+      vmImage : windows-2022
```
Comment (Iguins) [inline build/stage-build.yml:72]: Same here please use a global variable

File: build/stage-build.yml (hunk 21 above)
Comment (Iguins) [inline build/stage-build.yml:115]: please use a global variable like MacOSPoolname or something

File: build/stage-donetnew.yaml [hunk 32]
```diff
@@ -1,16 +1,14 @@
 jobs:
   - job: On_Windows_Dotnet_New
     pool:
-      name: $(windowsPoolName)
-    container: windows
-    
-    steps:    
+      vmImage : windows-2022
```
Comment (Iguins) [inline build/stage-donetnew.yaml:4]: Same here

File: build/variables.yml [hunk 33]
```diff
@@ -65,25 +65,25 @@ variables:
...
+  SolutionName: ApplicationTemplate
 
   # Paths to file
-  InfoPlistPath: $(Build.SourcesDirectory)/src/app/ApplicationTemplate.iOS/Info.plist
-  AndroidManifestPath: $(Build.SourcesDirectory)/src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml
+  InfoPlistPath: '$(Build.SourcesDirectory)/src/app/$(SolutionName).Mobile/iOS/Info.plist'
+  AndroidManifestPath: '$(Build.SourcesDirectory)/src/app/$(SolutionName).Mobile/Android/AndroidManifest.xml'
 
   # Pool names
   windowsPoolName: 'windows 2022'
```
Comment (Iguins) [inline build/variables.yml:75]: Those variables seems to be free to use :)

File: build/variables.yml [hunk 34]
```diff
@@ -65,25 +65,25 @@ variables:
...
   macOSPoolName: 'macOS'
 
   # Versions to use
-  NUGET_VERSION: 6.1.0
+  NUGET_VERSION: 6.2.0
   XAMARIN_IOS_VERSION: 15.10
```
Comment (Iguins) [inline build/variables.yml:80]: not needed anymore  as it was mac hosted dedicated only

File: build/templates/dotnet-install.yml [hunk 35]
```diff
@@ -0,0 +1,9 @@
+steps:
+  - task: UseDotNet@2
```
Comment (Iguins) [inline build/templates/dotnet-install.yml:2]: Do we really need to install a dependencies? Aren't they already loaded on the hosted agent?
@vincentcastagna

Comment (Soap-141) [reply build/templates/dotnet-install.yml:2]: Yeah the mobile workloads aren't install by default. I also made it like a MS engineer said to do for now on a developer blog.

https://devblogs.microsoft.com/dotnet/devops-for-dotnet-maui/

File: build/steps-build-ios.yml (hunk 26 above)
Comment (Soap-141) [reply build/steps-build-ios.yml:83]: I'm removing it for now since I wanted to do another Pull Request for cleaning and adding some variables like this.

**Review by takla21 — APPROVED** [review]
Good job!

File: src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml [hunk 36]
```diff
@@ -13,4 +13,4 @@
 			<meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/files" />
```
Comment (Iguins) [inline src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml:13]: Will this be remove in another PR?

Comment (Soap-141) [reply src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml:13]: Yes the whole Android Xamarin project, it's still there so we can compare in case something is missing or not working.

File: build/variables.yml [hunk 37]
```diff
@@ -65,25 +65,25 @@ variables:
...
   # Paths to file
-  InfoPlistPath: $(Build.SourcesDirectory)/src/app/ApplicationTemplate.iOS/Info.plist
-  AndroidManifestPath: $(Build.SourcesDirectory)/src/app/ApplicationTemplate.Droid/Properties/AndroidManifest.xml
+  InfoPlistPath: '$(Build.SourcesDirectory)/src/app/$(SolutionName).Mobile/iOS/Info.plist'
+  AndroidManifestPath: '$(Build.SourcesDirectory)/src/app/$(SolutionName).Mobile/Android/AndroidManifest.xml'
 
-  # Pool names
-  windowsPoolName: 'windows 2022'
+  # Virtual machine images
+  windowsVMImageName: 'windows-2022'
```
Comment (vincentcastagna) [inline build/variables.yml:75]: I would suggest to specify in the name that its hosted so its clear for everyone that we don't use our self-hosted agents. 

`windowsHostedAgentImage`:
`macOSHostedAgentImage`:

Comment (Soap-141) [reply build/variables.yml:75]: All right make sense!

File: src/app/ApplicationTemplate.Mobile/iOS/Info.plist [hunk 38]
```diff
@@ -0,0 +1,62 @@
...
+		<key>UIDeviceFamily</key>
+		<array>
+			<integer>1</integer>
+			<integer>2</integer>
+		</array>
+		<key>UILaunchStoryboardName</key>
+		<string>LaunchScreen</string>
+		<key>UIRequiredDeviceCapabilities</key>
+		<array>
+			<string>armv7</string>
```
Comment (vincentcastagna) [inline src/app/ApplicationTemplate.Mobile/iOS/Info.plist:24]: We shouldn't need to support armv7 anymore. Our iOS min target version is anyway to high for armv7 devices. See [here ](https://www.innerfence.com/howto/apple-ios-devices-dates-versions-instruction-sets)for more info.

**Review by vincentcastagna — APPROVED** [review]

**Review by Iguins — APPROVED** [review]
